data/transactions/logic/eval.go

```diff
@@ -2917,72 +2917,16 @@ func opEd25519VerifyBare(cx *EvalContext) error {
 	return nil
 }
 
-// leadingZeros needs to be replaced by big.Int.FillBytes
 func leadingZeros(size int, b *big.Int) ([]byte, error) {
 	data := b.Bytes()
 	if size < len(data) {
 		return nil, fmt.Errorf("insufficient buffer size: %d < %d", size, len(data))
 	}
```

Comment:
@ahangsu Sanity checking:  If I'm following the implementation, I'm wondering if it's now possible to avoid the copy introduced by `b.Bytes()`?

* I _think_  the _suggestion_ is equivalent to the PR.  The _suggestion_ counts the bytes in `b` for the if-statement.
* There might be more efficient or idiomatic ways to do the counting.  I looked briefly for alternatives.  Keep in mind that I'm mostly unfamiliar with Golang standard library.

```suggestion
	byteLength := int(math.Ceil(float64(b.BitLen()) / 8))

	if size < byteLength {
		return nil, fmt.Errorf("insufficient buffer size: %d < %d", size, byteLength)
	}
```

Reply:
I was lazy on this, but your suggestion works.

There is also a way of computing ceil-div-8 over fully integer: `(b.BitLen() + 7) / 8`, so long as the division rounds down to nearest integer.

Reply:
@ahangsu Thanks for bringing in the feedback via https://github.com/algorand/go-algorand/pull/3856/commits/83d361953899e10785f3027054bbf829333b8568.  Looks good - resolving.